### PR TITLE
refactor: fal file

### PR DESCRIPTION
**Pull request by badayvedat**

### CODE REVIEW

File: projects/fal/src/fal/toolkit/file/providers/gcp.py [hunk 1]
```diff
@@ -51,10 +51,13 @@ def bucket(self):
...
     def save(self, data: FileData) -> str:
-        destination_path = os.path.join(self.folder, data.file_name)
+        file_name = data.file_name or ""
+        destination_path = os.path.join(self.folder, file_name)
 
         gcp_blob = self.bucket.blob(destination_path)
-        gcp_blob.upload_from_string(data.data, content_type=data.content_type)
+
+        with data.data as file:
+            gcp_blob.upload_from_file(file, content_type=data.content_type)
```
Comment (mederka) [inline projects/fal/src/fal/toolkit/file/providers/gcp.py:60]: Will this work if filename == "" ?

**Review by mederka — APPROVED** [review]
Great job, @badayvedat 👏

As discussed. I think that maybe the class name File is a bit misleading because in reality it's not referring to a file but to a link to a file. It works magically when it's used inside a fal runtime, but not so in users local environment. I like your idea of potentially renaming this class to FileUrl or FileLink.

**Review by mederka — APPROVED** [review]
I say, once the tests pass, we ship it!

File: projects/fal/src/fal/toolkit/file/providers/gcp.py (hunk 1 above)
Comment (badayvedat) [reply projects/fal/src/fal/toolkit/file/providers/gcp.py:60]: good catch! it raises an error

**Pull request closed without merging**